flow/project.py

```diff
@@ -2443,9 +2459,60 @@ def _fetch_status(
         # setting self.config['flow']['status_parallelization']='none' if the argument
         # is True. But the later functionality will last the rest of the session but in order
         # to do proper deprecation, it is not required for now.
+        if status_parallelization not in ("thread", "process", "none"):
+            raise RuntimeError(
+                "Configuration value status_parallelization is invalid. "
+                "Valid choices are 'thread', 'process', or 'none'."
+            )
+
+        status_results = []
+
+        def compute_conditions(aggregate_id, aggregate, group, scheduler_status):
+            # Only single-operation groups are supported for status fetching
+            assert len(group.operations) == 1
+
+            status = {}
+            error_text = None
+            try:
+                status["scheduler_status"] = scheduler_status
+                completed = group._complete(aggregate)
+                status["completed"] = completed
+                eligible_ignore_conditions = (
+                    IgnoreConditions.NONE if completed else IgnoreConditions.POST
+                )
```

Comment:
When we print the status, we have to evaluate all pre/postconditions to determine both "eligible" (all preconditions met and some unmet postcondition) and "complete" (all postconditions met). With run/submit, we can exit early if some precondition is unmet / etc., but that is not the case for status (we must do more work). I found and implemented a partial optimization that will help in certain cases: If `completed=True` then the eligibility check is skipped. However, if `completed=False`, the eligibility check should ignore postconditions (it has already been established that `completed=False`, so checking postconditions is unnecessary). This is implemented in e0abac5.